command/meta.go

```diff
@@ -421,6 +421,21 @@ func (m *Meta) contextOpts() (*terraform.ContextOpts, error) {
 		}
 		opts.Providers = providerFactories
 		opts.Provisioners = m.provisionerFactories()
+
+		// If any dev overrides or unmanaged providers are in place, we cannot
+		// verify the locks as we have incomplete version information
+		if len(m.UnmanagedProviders) == 0 && len(m.ProviderDevOverrides) == 0 {
```

Comment:
If these settings (at all) mean this check doesn't happen, we should update docs in those areas (.. if there are docs areas?) to warn that this check won't happen if any unmanaged providers are present, or providerdevoverrides.

Reply:
And to your suggestions request ... you mention you'd like to filter. What about adding a method on `Locks` to allow this filtering? Since we have `addrs.Provider` in `UnmanagedProvider` or `ProviderDevOverrides`, could you use a new method to create a new `Locks` with those values filtered out?

Reply:
Unfortunately filtering development providers out of the locks won't work. The logic in context requires that there is an entry in the locks which satisfies the version constraints in the config.

To allow development providers to work while still verifying other providers, I think we'd need to pass a slice of providers to ignore through to the context, and then skip them like we do for builtin providers. There might be a cleaner way to do it, but I can't see one.

I guess in retrospect it's probably worth trying to do that, if only because it means I don't have to try to explain this weird edge case in the docs 🙃

Reply:
Done!